voluncheer/opportunityboard/views/search.py

```diff
@@ -0,0 +1,70 @@
+from django.shortcuts import render
+
+from opportunityboard.models import Category
+from opportunityboard.models import Opportunity
+from opportunityboard.models import Subcategory
+from opportunityboard.models import Subsubcategory
+from opportunityboard.views.opportunityboard import category_dict_gen
+
+
+def filter_search(request):
+    if request.method == "POST":
+        err, category = category_is_valid(request.POST["category"])
+        if err is not None:
+            print(err)
+        err, duration = duration_is_valid(request.POST["duration"])
+        if err is not None:
+            print(err)
+        err, distance = distance_is_valid(request.POST["distance"])
+        if err is not None:
+            print(err)
+
+    opportunity_lists = Opportunity.objects.order_by("-pubdate"[:20])
+    cate_output_dict = category_dict_gen()
+    durations = {
+        "One-day": ["<=2 Hours", "2-4 Hours", "Full Day"],
+        "Multi-day": ["3 Weeks", "1 Month", "2 Month", "6 Month"],
+    }
+    context = {
+        "opportunity_lists": opportunity_lists,
+        "categories": cate_output_dict,
+        "durations": durations,
+    }
+    return render(request, "voluncheer/opportunityboard.html", context)
+
+
+def category_is_valid(value):
+    """Check if the category is valid
+    Input: value[category_name] -> string
+    Return: error(None), target_object
+    """
+    category = Category.objects.filter(name=value).first()
+    if not category:
+        return None, category
+    subcategory = Subcategory.objects.filter(name=value).first()
+    if not subcategory:
+        return None, subcategory
+    subsubcategory = Subsubcategory.objects.filter(name=value).first()
+    if not subsubcategory:
+        return None, subsubcategory
+    return subsubcategory, None
+
+
+def duration_is_valid(value):
```

Comment:
The search errors out if you don't include a duration at all. Users may want to search for category without specifying a duration

Reply:
Also please don't leave code in that is either not in use or is commented out. Stash it locally or move it to another commit if you want to keep it around.

Here and below.